dali/base/dasds.cpp

```diff
@@ -1407,17 +1427,36 @@ class CDeltaWriter : implements IThreaded
         CriticalBlock b(pendingCrit);
         addToQueue(item);
     }
-    void clear()
+    void flush()
     {
-        CriticalBlock b(pendingCrit);
-        pending = {};
+        bool waitOnEmptySem = false;
+        {
+            CriticalBlock b(pendingCrit);
+            if (waiting)
```

Comment:
waiting is a very confusing variable name - especially since it is set from thread and cleared in another.  It actually seems to mean.

needToSendSignalToRequestDataIsWritten

I think it would be worth coming up with a clearer name - it would help check the logic is correct.

Reply:
I've renamed it deltaWriterIdle, which I think is more accurate and clarifies the logic IMO.